compiler/src/iree/compiler/Codegen/Common/DecomposePackUnPackOps.cpp

```diff
@@ -41,6 +75,23 @@ struct DecomposePackUnPackOpsPass
 void DecomposePackUnPackOpsPass::runOnOperation() {
   MLIRContext *ctx = &getContext();
   auto funcOp = getOperation();
+  {
+    RewritePatternSet patterns(ctx);
+    // Generalization patterns for outer unit dims have higher prirority because
+    // they do not generate reshape ops.
+    patterns.add<linalg::GeneralizeOuterUnitDimsPackOpPattern,
+                 linalg::GeneralizeOuterUnitDimsUnPackOpPattern>(
+        ctx,
+        /*benefit=*/10);
```

Comment:
Just pointing out, but not sure I have a suggestion here. Using pattern benefit is not a very stable way of making sure patterns apply in an order. Maybe have two separate `applyPatternsAndFoldGreedily` methods to have one trigger after another.

Reply:
I thought by design they are expected to apply patterns in an order if we assign benefits for all the patterns? The driver processes ops in a worklist-driven fashion and greedily applies the patterns that locally have the most benefit, while the benefits are assigned by us?

https://mlir.llvm.org/docs/PatternRewriter/#greedy-pattern-rewrite-driver

Reply:
That's the theory. It's not a very robust system. It's ok in this case (that's why I didn't push it) but better to not rely on ot

Reply:
I see, thanks!